packages/sync/src/offline/CordovaNetworkStatus.ts

```diff
@@ -0,0 +1,11 @@
+import {NetworkStatus} from "./NetworkStatus";
+
+export class CordovaNetworkStatus implements NetworkStatus {
+  public whenOnline(fn: any): void {
```

Comment:
Define callback function and use it everywhere. It could be any.

`public whenOnline(callback: OnlineCallback): `

IMHO `whenOffline` is missleading. How about registerOnlineHandler?
Alternatively we may need to have separate method `isOffline` or something like this to allow direct access to state. Not all components needs to listen.

Reply:
Actually "me thinking" when online/offline is kinda legacy inferace of the browser. 
Maybe we can just have single callback with the network state information like 
`
{
state: "online" | "offline",
metadata?: {
  type: "WIFI etc." 
}